### PR TITLE
[RFR] Added “Introducing Outline Audit”

### PR DESCRIPTION
**Pull request by HugoGiraudel**

### CODE REVIEW

File: _posts/2016-03-29-introducing-outline-audit.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+title: "Introducing Outline Audit"
+author: hugo-giraudel
+layout: post
+---
+
+We recently had some minor issues in a project where the final document outline on some pages was somewhat broken. I investigated a little, and ended up building [a tiny tool](https://github.com/edenspiekermann/outline-audit) to audit the document outline. More on that in a bit, but first…
+
+## What’s the document outline?
+
+The document outline is the theoretical schema constructed from the structure of a document (such as a web page), mostly based on headings. It matters because assistive technologies like screen readers heavily rely on this to give context and help navigation. It also matters for search engines, so they can locate and index meaningful content first.
```
Comment (nburkley) [inline _posts/2016-03-29-introducing-outline-audit.md]: "screen readers rely heavily on this" reads a bit better (they're both right though)

File: _posts/2016-03-29-introducing-outline-audit.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+## Does it still matter in HTML5?
+
+The short answer is yes.
+
+The long answer says that with HTML5, we got a set of new “sectioning elements”, such as `<header>`, `<footer>`, `<section>`, `<aside>`, `<article>` and so on. The theory (a.k.a the specification) says that inside one of these elements, the document outline is being resetted to create a sub-outline. The same way nested ordered list have their own counter and create sub-lists.
```
Comment (nburkley) [inline _posts/2016-03-29-introducing-outline-audit.md]: "document outline is being reset"

Comment (nburkley) [inline _posts/2016-03-29-introducing-outline-audit.md]: "The same way" => "In the same way, nested..."

File: _posts/2016-03-29-introducing-outline-audit.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+> If you as a developer want to provide a meaningful document structure, use the `<h1>`–`<h6>` elements to express document structure. DO NOT rely upon the HTML5 document outline. By all means use the HTML5 section elements, but do not rely upon them to convey a meaningful structure. If at some point in the future the HTML5 document outline ceases to be a fiction, you will be covered as the use of h1–h6 is backwards compatible.
+> — [Steve Faulkner](http://blog.paciellogroup.com/2013/10/html5-document-outline/)
+
+To avoid these issues, we can make our heading structure consistent and meaningful. Basically, it does not matter what kind of container we use, the heading level must be meaningful in regard to the previous heading. If it is a sort of sub-section, go down one level in the heading structure. If it is an unrelated section, have the same kind of heading as before. When checking the page, we should have a logical heading structure (which also mean never skipping a heading level). It’s actually what recommends the HTML5 specification:
```
Comment (nburkley) [inline _posts/2016-03-29-introducing-outline-audit.md]: "It’s actually what recommends the HTML5 specification" => "It’s actually what the HTML5 specification recommends"

File: _posts/2016-03-29-introducing-outline-audit.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+> Sections may contain headings of any rank, and authors are strongly encouraged to use headings of the appropriate rank for the section’s nesting level.
+> — [HTML 5.1 - Headings and sections](http://w3c.github.io/html/sections.html#headings-and-sections)
+
+Last thing to point out on this topic: the theory is so far away from the current state of things that the spec authors decided to review the whole thing to design something that will actually make sense and get implemented. [More to come in the next few months](https://github.com/w3c/html/issues/33). Meanwhile I highly recommend you read this [outstanding article about the document outline by Adrian Roselli](http://adrianroselli.com/2013/12/the-truth-about-truth-about-multiple-h1.html).
```
Comment (nburkley) [inline _posts/2016-03-29-introducing-outline-audit.md]: "Last thing to point out on this topic" - probably best to start with "The last thing" or "One last thing"

Comment (nburkley) [inline _posts/2016-03-29-introducing-outline-audit.md]: "review the whole thing to design something" => "review the whole thing and design something"

**Comment by martzoukos** [comment]
Nice!

File: _posts/2016-03-29-introducing-outline-audit.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Last, if you want to get the outline tree itself, you can use the `.get(..)` method:
+
+{% highlight js %}
+var outlineTree = o.get();
+{% endhighlight %}
+
+## How does it work?
+
+Nothing incredibly smart, really. It consists on a function that iterates over all headings of the document in one pass, building a data tree based on their position related to each others.
```
Comment (nburkley) [inline _posts/2016-03-29-introducing-outline-audit.md]: "Nothing incredibly smart" - maybe "Nothing incredibly complicated"?

**Comment by nburkley** [comment]
Great read, thanks @HugoGiraudel !

**Comment by HugoGiraudel** [comment]
Thanks. :)

**Pull request merged**